### PR TITLE
Add support for automatically add milestone

### PR DESCRIPTION
**Pull request by Nyholm**
This will add a milestone to a PR if all the following conditions are true: 

- The PR is not targeting the default branch
- The PR target has the same name as a milestone

If we always added a milestone, it would be hard to know the difference between "A milestone is selected properly" and "Milestone selected but might be wrong". So this will be forgiving for contributors to submit a PR to 5.x (default branch) but the PR was intended for 4.4. Ie, maintainers can be sure that if a milestone is set, it is correct.

### CODE REVIEW

File: tests/Controller/WebhookControllerTest.php [hunk 1]
```diff
@@ -58,6 +58,11 @@ public function getTests()
                 'pull_request.opened.json',
                 ['pull_request' => 3, 'status_change' => 'needs_review', 'pr_labels' => ['Bug']],
             ],
+            'On pull request opened with target branch' => [
+                'pull_request',
+                'pull_request.opened_target_branch.json',
+                ['pull_request' => 5, 'status_change' => 'needs_review', 'pr_labels' => ['Bug'], 'milestone'=>'4.4'],
```
Comment (Nyholm) [inline tests/Controller/WebhookControllerTest.php:64]: How can I test this properly?

File: src/Subscriber/MilestoneNewPRSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * Sets milestone on PRs that target non-default branch.
+     *
+     * @param GitHubEvent $event
+     */
+    public function onPullRequest(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        $repository = $event->getRepository();
+        if ('opened' !== $action = $data['action']) {
```
Comment (weaverryan) [inline src/Subscriber/MilestoneNewPRSubscriber.php:34]: `$action` variable not needed here

File: src/Issues/GitHub/MilestonesApi.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        $this->issuesApi->update($repository->getVendor(), $repository->getName(), $issueNumber, [
+            'milestone' => $milestoneNumber,
+        ]);
+    }
+
+    /**
+     * @return bool
+     */
+    public function exits(Repository $repository, string $milestoneName)
```
Comment (weaverryan) [inline src/Issues/GitHub/MilestonesApi.php:73]: I think this should be `exists()`. You actually consistently named it, so I think the code *would* work as is ;)

File: tests/Controller/WebhookControllerTest.php (hunk 1 above)
Comment (weaverryan) [reply tests/Controller/WebhookControllerTest.php:64]: Not really a great way - it's a very basic "smoke test" :/

**Review by weaverryan — COMMENTED** [review]
Minor tweaks!

File: src/Issues/GitHub/MilestonesApi.php (hunk 3 above)
Comment (Nyholm) [reply src/Issues/GitHub/MilestonesApi.php:73]: oops =)

File: src/Subscriber/MilestoneNewPRSubscriber.php (hunk 2 above)
Comment (Nyholm) [reply src/Subscriber/MilestoneNewPRSubscriber.php:34]: Thank you

**Comment by Nyholm** [comment]
Thank you for the review. I've updated the RP

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**